types/jsonwebtoken/index.d.ts

```diff
@@ -159,12 +160,12 @@ export function sign(
  * callback - Callback to get the encoded token on
  */
 export function sign(
-    payload: string | Buffer | object,
+    payload: string | Buffer | { [key: string]: any },
```

Comment:
```suggestion
    payload: string | Buffer | { [key: string]: unknown },
```

I would recommend "unknown" instead of "any".

The advantage of "unknown" is that the type has to be tested before you can work with it.